src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/RazorSemanticTokenInfoServiceTest.cs

```diff
@@ -899,6 +1076,64 @@ private static Range GetRange(string text)
         return range;
     }
 
+    private async Task CompareResultsToWhenFeatureFlagTurnedOffAsync(
+        string documentText,
+        bool isRazorFile,
+        Range razorRange,
+        Dictionary<Range, int[]> perRangeTokens)
+    {
+        if (!UsePreciseSemanticTokenRanges)
+        {
+            var singleRangeResponse = Assert.Single(perRangeTokens);
+            Assert.True(singleRangeResponse.Value.Count() % 5 == 0);
+            return;
+        }
+
+        UsePreciseSemanticTokenRanges = false;
+        var (originalCsResponse, originalCsTokens) = await GetCSharpSemanticTokensResponseAsync(documentText, razorRange, isRazorFile);
```

Comment:
Can you comment what this additional verification is is doing? Hasn't this already been called? Isn't that where the `perRangeTokens` came from? What am I missing here?